packages/react/__tests__/custom-cache.js

```diff
@@ -38,3 +39,41 @@ test('with custom plugins', () => {
     />
   `)
 })
+
+test('with custom sheet', () => {
+  // https://github.com/emotion-js/emotion/issues/2675
+  let cache = createCache({
+    key: 'test',
+    speedy: false
+  })
+
+  class GlobalSheet extends StyleSheet {
+    insert(rule) {
+      return super.insert('/** global */' + rule)
+    }
+  }
+  class MySheet extends StyleSheet {
+    constructor(options) {
+      super(options)
+      if (Reflect.get(options.container, 'sheet')) {
+        return new GlobalSheet(options)
+      } else {
+        Reflect.set(options.container, 'sheet', this)
+      }
```

Comment:
I wonder - any particular reason why `Reflect` is used here?

Reply:
To bypass the flow type check. `sheet` is not a property on the `HTMLElement`